File: drills/python/harnesses/nc99_distinct_subsequences.py
try:
    (numDistinct)
except NameError:
    raise Exception("__signature_mismatch__")

__results__ = []
def __case__(label, expected, actual):
    __results__.append([label, repr(expected), repr(actual)])

__case__("numDistinct('rabbbit', 'rabbit')", 3, numDistinct('rabbbit', 'rabbit'))
__case__("numDistinct('babgbag', 'bag')", 5, numDistinct('babgbag', 'bag'))
__case__("numDistinct('', 'a')", 0, numDistinct('', 'a'))
__case__("numDistinct('a', '')", 1, numDistinct('a', ''))
__case__("numDistinct('abc', 'abc')", 1, numDistinct('abc', 'abc'))
__case__("numDistinct('aaa', 'aa')", 3, numDistinct('aaa', 'aa'))

File: drills/elixir/harnesses/nc99_distinct_subsequences.exs
[
  {~S{num_distinct("rabbbit", "rabbit")}, inspect(3), inspect(Solution.num_distinct("rabbbit", "rabbit"))},
  {~S{num_distinct("babgbag", "bag")}, inspect(5), inspect(Solution.num_distinct("babgbag", "bag"))},
  {~S{num_distinct("", "a")}, inspect(0), inspect(Solution.num_distinct("", "a"))},
  {~S{num_distinct("a", "")}, inspect(1), inspect(Solution.num_distinct("a", ""))},
  {~S{num_distinct("abc", "abc")}, inspect(1), inspect(Solution.num_distinct("abc", "abc"))},
  {~S{num_distinct("aaa", "aa")}, inspect(3), inspect(Solution.num_distinct("aaa", "aa"))}
]
